### PR TITLE
Moan about old Pull Requests

### PR DESCRIPTION
**Pull request by jonnyarnold**
When `mix do app.start, daily_checks` is run, Samuel will look at all open Pull Requests. If the number of days the PR has been open is divisible by 7 (and isn't 0), Samuel will comment.

This _should_ work with [Heroku Scheduler](https://elements.heroku.com/addons/scheduler).

### CODE REVIEW

**Comment by reevoo-samuel** [comment]
*If you're new, read the [guidelines](https://github.com/reevoo/guidelines/blob/master/pull_requests.md).*

Good code is **tested**, **designed for change**, **easy to understand** 
and **able to cope when things go wrong**.

Good pull requests are **descriptive**, **small** and **short-lived**.

**For the Author**
- [ ] Description explains what you have done and why.
- [ ] Tests have been written (or you have explained why you don't need
tests).
- [ ] Documentation/comments have been added/updated.

**For the Reviewer**
- [ ] The code is well-designed and able to cope with change.
- [ ] The code is easy to understand.
- [ ] Tests cover happy and unhappy paths in the code.
- [ ] Tests are green on CI (or pulled/tested on your machine).

File: config/copy_config.exs [hunk 1]
```diff
@@ -2,9 +2,10 @@ use Mix.Config
 
 config :samuel,
```
Comment (lpil) [inline config/copy_config.exs]: Why does this stuff live in config again?

File: config/github_config.exs [hunk 2]
```diff
@@ -5,5 +5,11 @@ github_access_key = case Mix.env do
   otherwise -> System.get_env "GITHUB_ACCESS_KEY"
 end
 
+org_name = case Mix.env do
```
Comment (lpil) [inline config/github_config.exs]: I think it's more idomatic to have another file for test, etc. Doesn't really matter, but maybe we'll change this later.

File: config/copy_config.exs (hunk 1 above)
Comment (jonnyarnold) [inline config/copy_config.exs]: If other companies want to use it, they can choose their own messages.

File: lib/samuel/checks/has_comments.ex [hunk 3]
```diff
@@ -19,8 +19,7 @@ defmodule Samuel.Checks.HasComments do
   def action(event) do
     %Action{
       action: :post_comment,
-      repo: event["repository"]["full_name"],
-      pull_id: event["pull_request"]["number"],
+      comments_url: event["pull_request"]["comments_url"],
```
Comment (lpil) [inline lib/samuel/checks/has_comments.ex]: Why did we lose these?

File: lib/mix/tasks/daily_checks.ex [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  use Mix.Task
+  alias Samuel.Events
+
+  def run(_args) do
+    Events.respond(tick_event)
+  end
+
+  defp tick_event do
+    %{
+      "action" => "daily"
```
Comment (lpil) [inline lib/mix/tasks/daily_checks.ex]: I find the fact we're building up what looks a bit like the github webhook here a bit strange.

File: lib/samuel/check.ex [hunk 5]
```diff
@@ -15,16 +15,6 @@ defmodule Samuel.Check do
   """
   defcallback check(pull_request_data)  :: none | action
```
Comment (lpil) [inline lib/samuel/check.ex]: `defcallback` has been deprecated, use `@callback`

http://elixir-lang.org/docs/stable/elixir/Behaviour.html

File: lib/samuel/checks/old.ex [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  def check(data) do
+    data.open_prs["items"]
+    |> Enum.map(fn(pr) -> check(pr, data) end)
+    |> Enum.reject(fn(action) -> action == nil end)
+  end
+
+  defp check(pr, data) do
+    days = days_since_creation(pr)
+
+    if days != 0 && rem(days, 7) == 0 do
```
Comment (lpil) [inline lib/samuel/checks/old.ex]: So we reply once a week?

File: lib/samuel/events.ex [hunk 7]
```diff
@@ -0,0 +1,18 @@
+defmodule Samuel.Events do
```
Comment (lpil) [inline lib/samuel/events.ex]: Module names are generally singular. Oops.

**Comment by lpil** [comment]
I feel like I want to try and re-write this app, it all feels a bit coupled to the web component, which should really just be a plugin the rest of the app is blind to.

**Comment by lpil** [comment]
But yeah, ship it

File: config/copy_config.exs (hunk 1 above)
Comment (lpil) [inline config/copy_config.exs]: We've not really given them a good mechanism to do this here I think

File: lib/samuel/checks/old.ex (hunk 6 above)
Comment (jonnyarnold) [inline lib/samuel/checks/old.ex]: Yep :)

**Comment by jonnyarnold** [comment]
I agree with you that it would be good to have another go at this.

SHIPPING ANYWAY

**Comment by lpil** [comment]
In the spirit of Samuel we are ignoring all content of the review and hitting that merge button ;)

**Comment by jonnyarnold** [comment]
Party Parrot

**Pull request merged**